stdlib/runtime/numbers.gr

```diff
@@ -1070,14 +1070,14 @@ export let (>) = (x: Number, y: Number) => {
 export let numberGreater = (>);
 
 export let (<=) = (x: Number, y: Number) => {
-  let x = WasmI32.fromGrain(x)
-  let y = WasmI32.fromGrain(y)
   // Equality is finicky, so delegate
-  let xval = coerceFloat64(x)
-  let yval = coerceFloat64(y)
+  let xval = coerceNumberToWasmF64(x)
+  let yval = coerceNumberToWasmF64(y)
   if (WasmF64.lt(xval, yval)) {
     true
   } else {
+    let x = WasmI32.fromGrain(x)
+    let y = WasmI32.fromGrain(y)
```

Comment:
This is technically doing a `fromGrain` twice in the `else` case?

Reply:
What do you mean?

Reply:
the coerce call is made at the top of this function, which does a `fromGrain` inside that function, then when we hit this else case, we are calling fromGrain again.